browser/ui/brave_actions/brave_action_view_controller.cc

```diff
@@ -87,19 +87,44 @@ ui::MenuModel* BraveActionViewController::GetContextMenu(
   return nullptr;
 }
 
-bool BraveActionViewController::ExecuteActionUI(
-    std::string relative_path) {
-  return TriggerPopupWithUrl(PopupShowAction::SHOW_POPUP,
-      extension()->GetResourceURL(relative_path),
-      true);
+void BraveActionViewController::ExecuteActionUI(
+    const std::string& relative_path) {
+  TriggerPopupWithUrl(PopupShowAction::SHOW_POPUP,
+                      extension()->GetResourceURL(relative_path),
+                      /*grant_tab_permissions=*/true);
 }
 
 ExtensionActionViewController*
 BraveActionViewController::GetPreferredPopupViewController() {
   return this;
 }
 
-bool BraveActionViewController::TriggerPopupWithUrl(
+void BraveActionViewController::TriggerPopup(PopupShowAction show_action,
+                                             bool grant_tab_permissions) {
+  content::WebContents* const web_contents =
+      view_delegate_->GetCurrentWebContents();
+  const int tab_id = sessions::SessionTabHelper::IdForTab(web_contents).id();
+
+  TriggerPopupWithUrl(show_action, extension_action_->GetPopupUrl(tab_id),
+                      grant_tab_permissions);
+}
+
+void BraveActionViewController::OnPopupClosed() {
```

Comment:
in the future if possible please try to add methods in a way that doesn't cause git to rearrange them in the diff. It's much harder to tell what the real differences are